lib/netext/httpext/request_test.go

```diff
@@ -280,3 +284,69 @@ func BenchmarkWrapDecompressionError(b *testing.B) {
 		_ = wrapDecompressionError(err)
 	}
 }
+
+func TestTrailFailed(t *testing.T) {
+	t.Parallel()
+	srv := httptest.NewTLSServer(httpbin.New().Handler())
+	defer srv.Close()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	samples := make(chan stats.SampleContainer, 10)
+	logger := logrus.New()
+	logger.Level = logrus.DebugLevel
+	state := &lib.State{
+		Options: lib.Options{
+			RunTags:    &stats.SampleTags{},
+			SystemTags: &stats.DefaultSystemTagSet,
+		},
+		Transport: srv.Client().Transport,
+		Samples:   samples,
+		Logger:    logger,
+		BPool:     bpool.NewBufferPool(2),
+	}
+	ctx = lib.WithState(ctx, state)
+
+	testCases := map[string]struct {
+		responseCallback func(int) bool
+		failed           null.Bool
+	}{
+		"null responsecallback": {responseCallback: nil, failed: null.NewBool(false, false)},
+		"unexpected response":   {responseCallback: func(int) bool { return false }, failed: null.NewBool(true, true)},
+		"expected response":     {responseCallback: func(int) bool { return true }, failed: null.NewBool(false, true)},
+	}
+	for name, testCase := range testCases {
+		responseCallback := testCase.responseCallback
+		failed := testCase.failed
+
+		t.Run(name, func(t *testing.T) {
+			ctx = lib.WithState(ctx, state)
```

Comment:
Do you need to reattach the state here? It's already done on line 307.

Reply:
no, I probably just copy-pasted more when I was moving to a testcases variant